source/contracts/reporting/ReportingWindow.sol

```diff
@@ -218,6 +228,20 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable, IReportingWi
         return true;
     }
 
+    function collectReportingFees(address _reporterAddress, uint256 _attoReportingTokens) public returns (bool) {
+        IReportingToken _shadyReportingToken = IReportingToken(msg.sender);
+        require(isContainerForReportingToken(_shadyReportingToken));
+        // NOTE: Will need to handle other denominations when that is implemented
+        ICash _cash = ICash(controller.lookup("Cash"));
+        uint256 _balance = _cash.balanceOf(this);
+        uint256 _feePayoutShare = _balance.mul(_attoReportingTokens).div(totalWinningReportingTokens);
+        if (_feePayoutShare > 0) {
+            _cash.withdrawEtherTo(_reporterAddress, _feePayoutShare);
+        }
+        totalWinningReportingTokens = totalWinningReportingTokens.sub(_attoReportingTokens);
```

Comment:
Security: Always update state _before_ calling out to external contracts (`.withdrawEtherTo` will call an external contract).  At the moment, this is subject to a reentrancy attack.

Reply:
Good catches. Will do